src/sql/reuse/mod.rs

```diff
@@ -0,0 +1,71 @@
+use nom_sql::Table;
+use sql::query_graph::QueryGraph;
+use mir::MirQuery;
+
+use std::vec::Vec;
+use std::collections::HashMap;
+
+mod finkelstein;
+mod relaxed;
+mod full;
+mod helpers;
+
+#[derive(Clone, Debug)]
+pub enum ReuseType {
+    DirectExtension,
+    PrefixReuse,
+    #[allow(dead_code)]
+    BackjoinRequired(Vec<Table>),
+}
+
+enum ReuseConfigType {
+    Finkelstein,
+    Relaxed,
+    Full,
+}
+
+const REUSE_CONFIG: ReuseConfigType = ReuseConfigType::Finkelstein;
```

Comment:
I guess we'll eventually want to turn this into a dynamic option passed when creating a `SqlIncorporator`, rather than a compile-time constant, but this totally fine for now.